app/model/repository/Comments.php

```diff
@@ -31,4 +32,57 @@ public function persistComment(Comment $comment) {
   public function persistThread(CommentThread $thread) {
     $this->persist($thread);
   }
+
+  /**
+   * Get the number of visible comments for given user
+   * @param $threadId Id of the comments thread.
+   * @param $userId Id of the viewer or null (if nobody is logged in).
+   * @param bool $allVisible True = all coments visible for the user, false = comments made by the user.
+   */
+  public function getThreadCommentsCount(CommentThread $thread, ?User $user, bool $allVisible = true)
+  {
+    if (!$user) {
+      return 0; // anonymous users cannot see any comments
+    }
+
+    $qb = $this->comments->createQueryBuilder('tc')
+      ->select('COUNT(tc.id)')
+      ->where('tc.commentThread = :id');
+
+    if ($allVisible) {
+      $qb->andWhere('tc.user = :user OR tc.isPrivate = 0');
+    } else {
+      $qb->andWhere('tc.user = :user');
+    }
+
+    $qb->setParameters([
+      'id' => $thread->getId(),
+      'user' => $user->getId(),
+    ]);
+
+    return $qb->getQuery()->getSingleScalarResult();
+  }
+
+  /**
+   * Get the number of visible comments for given user
+   * @param $threadId Id of the comments thread.
+   * @param $userId Id of the viewer or null (if nobody is logged in).
+   */
+  public function getThreadLastComment(CommentThread $thread, ?User $user)
+  {
+    if (!$user) {
+      return null; // anonymous users cannot see any comments
```

Comment:
I would probably check this before calling the method and request a non-null $user, but feel free to dismiss this as a matter of personal taste.

Reply:
It is also checked before call. This is just a paranoid preemptive measure.

Reply:
Then change the type in the signature and let the type system do the work for you :-)

Reply:
That was not the original intent, but very well, I do not mind.